src/components/form/Form.tsx

```diff
@@ -1,79 +1,52 @@
 'use client';
 
 import { zodResolver } from '@hookform/resolvers/zod';
-import { useTranslations } from 'next-intl';
 import type { ReactNode } from 'react';
-import { useEffect } from 'react';
 import type {
   DefaultValues,
 } from 'react-hook-form';
 import {
   FormProvider,
   useForm,
 } from 'react-hook-form';
-import type { z } from 'zod';
-
-import ErrorSummary from '@/ui/error-summary/error-summary';
+import type { ZodSchema } from 'zod';
 
 export type FormProps<T extends object> = {
   defaultValues?: DefaultValues<T>;
-  schema?: z.Schema<DefaultValues<T>>;
-  isSubmitting?: boolean;
-  action: (values: T) => Promise<unknown>;
-  errors?: {
-    [key: string]: unknown;
-  };
+  schema?: ZodSchema;
+  action: (values: T) => Promise<{ data?: unknown; error?: object | string; ok: boolean }>;
   i18path: keyof IntlMessages['Forms'];
   children: ReactNode;
+  successMessage?: ReactNode;
 };
 
 const Form = <T extends object>({
   defaultValues,
   schema,
-  errors: responseErrors,
   children,
   action,
-  i18path,
 }: FormProps<T>) => {
-  const t = useTranslations(`Forms.${i18path}`);
-
   const methods = useForm({
     defaultValues,
     resolver: schema ? zodResolver(schema) : undefined,
   });
-  const {
-    formState: { errors },
-  } = methods;
 
-  useEffect(() => {
-    if (responseErrors) {
-      for (const [key, value] of Object.entries(responseErrors)) {
-        methods.setError(`root.${key}`, {
-          type: 'custom',
-          message: {
-            summary: value,
-            input: value,
-          } as any,
-        });
+  const onSubmit = async (values: T) => {
+    const { ok, error } = await action(values);
+    if (!ok) {
```

Comment:
To avoid nesting `ifs` we can make something like
```
if (ok) {
return;
}
```

It is more readable for me. What do you think?

Reply:
I will remove this component at all